script/model/execution_engine_runner/generate_scan.py

```diff
@@ -48,24 +64,30 @@ def generate_main_fun(fun_names):
     print("}")
 
 
-def generate_all():
+def generate_all(include_output):
     fun_names = []
     col_nums = range(1, 16, 2)
     row_nums = [1, 3, 5, 7, 10, 50, 100, 500, 1000, 2000, 5000, 10000, 20000, 50000, 100000, 200000, 500000, 1000000]
 
     generate_state()
 
+    if include_output == 1:
+        generate_output_struct(16)
+
     for col_num in col_nums:
         for row_num in row_nums:
             cardinalities = [1]
             while cardinalities[-1] < row_num:
                 cardinalities.append(cardinalities[-1] * 2)
             cardinalities[-1] = row_num
             for cardinality in cardinalities:
-                fun_names.append(generate_scan_fun(col_num, row_num, cardinality))
+                fun_names.append(generate_scan_fun(col_num, row_num, cardinality, include_output))
 
     generate_main_fun(fun_names)
 
 
 if __name__ == '__main__':
-    generate_all()
+    aparser = argparse.ArgumentParser(description='Mini Trainer')
+    aparser.add_argument('--include_output', type=int, default=0, help='Test data split ratio')
```

Comment:
nit: the help text should probably be about whether to write to the OutputBuffer?